### PR TITLE
Update Kubernetes and kind version

### PR DESCRIPTION
**Pull request by constanca-m**
## What does this PR do?
Adds support for Kubernetes 1.26.0 version and updates Kind version

## Related issues

- Relates to https://github.com/elastic/observability-dev/issues/2345

### CODE REVIEW

File: .ci/scripts/install-kubernetes-autodiscover-test-dependencies.sh [hunk 1]
```diff
@@ -16,8 +16,8 @@ set -euxo pipefail
 MSG="parameter missing."
 HOME=${HOME:?$MSG}
 
-KIND_VERSION="v${KIND_VERSION:-"0.14.0"}"
-KUBERNETES_VERSION="${KUBERNETES_VERSION:-"1.25.0"}"
+KIND_VERSION="v${KIND_VERSION:-"0.17.0"}"
+KUBERNETES_VERSION="${KUBERNETES_VERSION:-"1.27.0"}"
```
Comment (gizas) [inline .ci/scripts/install-kubernetes-autodiscover-test-dependencies.sh:20]:
```suggestion
KUBERNETES_VERSION="${KUBERNETES_VERSION:-"1.26.0"}"
```

File: e2e/_suites/kubernetes-autodiscover/README.md [hunk 2]
```diff
@@ -110,8 +110,8 @@ This is an example of the optional configuration:
    export BEAT_VERSION=7.12.0 # version of beats to use
    export ELASTIC_AGENT_VERSION=7.12.0 # version of Elastic Agent to use
    export GITHUB_CHECK_SHA1=0123456789 # to select snapshots built by beats-ci
-   export KIND_VERSION="0.14.0" # version of kind
-   export KUBERNETES_VERSION="1.25.0" # version of the cluster to be passed to kind
+   export KIND_VERSION="0.17.0" # version of kind
+   export KUBERNETES_VERSION="1.27.0" # version of the cluster to be passed to kind
```
Comment (gizas) [inline e2e/_suites/kubernetes-autodiscover/README.md:114]:
```suggestion
   export KUBERNETES_VERSION="1.26.0" # version of the cluster to be passed to kind
```

File: internal/config/kubernetes/README.md [hunk 3]
```diff
@@ -3,8 +3,8 @@
 ## Requirements
 
 - docker
-- kind (>= 0.14.0)
-- kubectl (>= 1.25.0)
+- kind (>= 0.17.0)
+- kubectl (>= 1.27.0)
```
Comment (gizas) [inline internal/config/kubernetes/README.md:7]:
```suggestion
- kubectl (>= 1.26.0)
```

**Review by gizas — APPROVED** [review]

**Review by tetianakravchenko — APPROVED** [review]
the failing test is an `fleet_ubuntu_22_04_amd64_upgrade_agent`, it is a known issue
I would suggest to merge this PR, as changes are not related to the failing test

**Pull request merged**